controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -109,3 +246,32 @@ func (r *StorageClassClaimReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&ocsv1alpha1.StorageClassClaim{}).
 		Complete(r)
 }
+
+func (r *StorageClassClaimReconciler) get(ctx context.Context, obj client.Object) error {
+	key := client.ObjectKeyFromObject(obj)
+	return r.Client.Get(ctx, key, obj)
+}
+
+func (r *StorageClassClaimReconciler) updateStatus(ctx context.Context, obj client.Object, state ocsv1alpha1.PhaseOperation) {
+	r.storageClassClaim.Status.Phase = state
+	statusErr := r.Client.Status().Update(ctx, obj)
+	if statusErr != nil {
+		r.log.Error(statusErr, "Failed to update "+obj.GetObjectKind().GroupVersionKind().Kind+" status.")
+	}
+}
+
+func (r *StorageClassClaimReconciler) delete(ctx context.Context, obj client.Object) error {
+	if err := r.Client.Delete(ctx, obj); err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	return nil
+}
```

Comment:
This is used only once and has a special logic that is not standard for k8s delete. 
The reader will benefit from having this logic embedded in the call site itself